### ISSUE TITLE
After webpack hot reload, update function effects for `PageView` action happen twice

### ISSUE THREAD

**Issue by ghost** [alexmingoia/purescript-pux#77]
As the title says, for some reason after the hot reload takes place, the effects returned by my `update` function happens twice. And it seems like it's only for the `PageView` action.

To reproduce...
1. Clone the pux-starter-app repo
2. Update to webpack 2
3. Change the `update` function to use `EffModel`, add a couple routes, and add some `link` to the Counter view so we can change between pages.
4. Add a console log statement to the pageview update function
5. Navigate between pages. The `console.log` fires once per page view, great. 
6. Now modify some text in the view to get webpack to recompile and hot reload. 
7. Now click the links again to switch between pages, and the `console.log` messages fire twice each time! If you hot-reload a third time, now each page view triggers 3 console logs... and so on. 

From a fresh project and an update function is as simple as this...

```
update (PageView route) state =
  { state: state { route = route }
  , effects: [ (liftEff $ log "pageview action" >>= (\_ -> pure NoOp)) ]
  }
```

It _only_ doubles the effects of the PageView actions, so it's something related to the popstate handler / history API, but as far as I can tell the javascript handlers in `Router.js` are only being run once.

Am I doing something seriously wrong or did I find a bug? Could it be something to do with webpack 2? 

(I can upload a GitHub repo that reproduces the issue)

**Comment by ghost** [comment]
To make it even freakier, there are no extra states in the Pux Devtool history. For 3 page views -> 3 state changes in the devtool (6 if we're counting the NoOp actions).

**Comment by alexmingoia** [comment]
The redundant initial renders have been fixed in Pux 8.7
